core/src/test/java/org/infinispan/api/APINonTxTest.java

```diff
@@ -661,4 +677,99 @@ public boolean equals(Object obj) {
       }
    }
 
+   void assertLockStream(BiConsumer<Cache<Object, Object>, CacheEntry<Object, Object>> consumer) throws Throwable {
+      for (int i = 0; i < 10; i++) {
+         cache.put(i, "value" + i);
+      }
+
+      CyclicBarrier barrier = new CyclicBarrier(2);
+
+      int key = 4;
+
+      LockedStream<Object, Object> stream = cache.getAdvancedCache().lockedStream();
+      Future<?> forEachFuture = fork(() -> stream.forEach((c, e) -> {
+         if (e.getKey().equals(key)) {
+            try {
+               barrier.await(10, TimeUnit.SECONDS);
```

Comment:
Maybe I've missed that during my last review but I don't think this test is doing what you want. Here, the consumer and the put that's about to timeout are running concurrently. I suppose that you want actually run two puts that time out, one before consumer invocation and another afterwards, to verify that the command did not unlock the lock:
```java
      Future<?> forEachFuture = fork(() -> stream.forEach((c, e) -> {
         if (e.getKey().equals(key)) {
            try {
               barrier.await(10, TimeUnit.SECONDS);
               barrier.await(10, TimeUnit.SECONDS);
               consumer.accept(c, e);
               barrier.await(10, TimeUnit.SECONDS);
               barrier.await(10, TimeUnit.SECONDS);
            } catch (InterruptedException | BrokenBarrierException | TimeoutException e1) {
               throw new RuntimeException(e1);
            }
         }
      }));
      barrier.await(10, TimeUnit.SECONDS);
      Future<Object> putFutureBefore = fork(() -> cache.put(key, "value" + key + "-new"));
      Exceptions.expectException(TimeoutException.class, () -> putFutureBefore.get(50, TimeUnit.MILLISECONDS));
      barrier.await(10, TimeUnit.SECONDS);
 
      barrier.await(10, TimeUnit.SECONDS);
      barrier.await(10, TimeUnit.SECONDS);
    
      Future<Object> putFutureAfter = fork(() -> cache.put(key, "value" + key + "-new"));
      Exceptions.expectException(TimeoutException.class, () -> putFutureAfter.get(50, TimeUnit.MILLISECONDS));
      barrier.await(10, TimeUnit.SECONDS);

      forEachFuture.get(10, TimeUnit.SECONDS);
```